cmd/witr/main.go

```diff
@@ -15,252 +15,210 @@ import (
 	"github.com/pranshuparmar/witr/internal/source"
 	"github.com/pranshuparmar/witr/internal/target"
 	"github.com/pranshuparmar/witr/pkg/model"
+	"github.com/spf13/cobra"
 )
 
-var version = ""
-var commit = ""
-var buildDate = ""
-
-func printHelp() {
-	fmt.Println("Usage: witr [--pid N | --port N | name] [--short] [--tree] [--json] [--warnings] [--no-color] [--env] [--help] [--version]")
-	fmt.Println("  --pid <n>         Explain a specific PID")
-	fmt.Println("  --port <n>        Explain port usage")
-	fmt.Println("  --short           One-line summary")
-	fmt.Println("  --tree            Show full process ancestry tree")
-	fmt.Println("  --json            Output result as JSON")
-	fmt.Println("  --warnings        Show only warnings")
-	fmt.Println("  --no-color        Disable colorized output")
-	fmt.Println("  --env             Show only environment variables for the process")
-	fmt.Println("  --help            Show this help message")
-	fmt.Println("  --version         Show version and exit")
-}
-
-// Helper: which flags need a value (not bool flags)?
-func flagNeedsValue(flag string) bool {
-	switch flag {
-	case "--pid", "-pid", "--port", "-port":
-		return true
-	}
-	return false
-}
+var (
+	version   = ""
+	commit    = ""
+	buildDate = ""
+)
 
 func main() {
 	// Sanity check: fail build if version is not injected
 	if version == "" {
 		fmt.Fprintln(os.Stderr, "ERROR: version not set. Use -ldflags '-X main.version=...' when building.")
 		os.Exit(2)
 	}
```

Comment:
I think, there is no need to prevent witr to run if there is no version specified.

```suggestion
    if version == "" {
      version = "dev"
    }
```